src/Sonarr.Api.V3/ProviderControllerBase.cs

```diff
@@ -89,6 +89,12 @@ public ActionResult<TProviderResource> CreateProvider([FromBody] TProviderResour
         public ActionResult<TProviderResource> UpdateProvider([FromBody] TProviderResource providerResource, [FromQuery] bool forceSave = false)
         {
             var existingDefinition = _providerFactory.Find(providerResource.Id);
+
+            if (existingDefinition == null)
```

Comment:
To your point it'd be worth checking for the ID in the route first, then the body, in API v4 we could drop the fallback.